### PR TITLE
Extract varargs' related logic into a separate package.

### PR DESCRIPTION
**Pull request by immutableT**
Extract varargs' related logic into a separate package.

### CODE REVIEW

File: internal/pkg/varargs/varargs.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	}
+
+	a, ok := sl.X.(*ssa.Alloc)
+	if !ok || a.Comment != "varargs" {
+		return nil
+	}
+
+	var (
+		stores []*ssa.Store
+	)
```
Comment (PurelyApplied) [inline internal/pkg/varargs/varargs.go:53]: Nit: this should be one line.

File: internal/pkg/varargs/varargs.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+	for _, r := range *a.Referrers() {
+		idx, ok := r.(*ssa.IndexAddr)
+		if !ok {
+			continue
+		}
+
+		if idx.Referrers() != nil && len(*idx.Referrers()) != 1 {
```
Comment (PurelyApplied) [inline internal/pkg/varargs/varargs.go:61]: In what care would `len(*idx.Referrers())` be greater than one?  Should we have test coverage for that case?

File: internal/pkg/varargs/varargs.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+// ReferredBy determines if the supplied node refers the Vararg in question via a store instruction.
+func (v *Varargs) ReferredBy(r referrer) bool {
+	for _, s := range v.stores {
+		if r.HasPathTo(s.Val.(ssa.Node)) {
+			return true
+		}
+	}
+
+	return false
+}
```
Comment (PurelyApplied) [inline internal/pkg/varargs/varargs.go:89]: The unit tests do not cover the `return false` case, though the end-to-end `levee_test.go` does.  Should we cover that here?

File: internal/pkg/varargs/varargs.go (hunk 2 above)
Comment (immutableT) [reply internal/pkg/varargs/varargs.go]: Thanks for point this out - this has been bothering me.
I don't think there is such a case, and if there is one, I would rather panic and find out.
Removed.

File: internal/pkg/varargs/varargs.go (hunk 3 above)
Comment (immutableT) [reply internal/pkg/varargs/varargs.go:83]: Added test coverage.

File: internal/pkg/varargs/varargs.go (hunk 1 above)
Comment (immutableT) [reply internal/pkg/varargs/varargs.go]: Done.

**Pull request merged**